### PR TITLE
Add package: proxychains-ng

### LINKED ISSUE #330: Package request: proxychains-ng
I often need to SSH through socks5 proxies and proxychains is my favorite for that. Planning to look into building the package myself. I tried to directly compile from termux but that didn't work :P

Upstream repo: https://github.com/rofl0r/proxychains-ng

### PR DESCRIPTION
**Pull request by Neo-Oli**
Closes #330 (#395)

I've tested it with a simple ssh tunnel and it seems to work really
well. It works better than tsocks because I was able use python
programs (getmail) with this. I hope it holds up when more experienced
users put it through it's paces.

### CODE REVIEW

**Comment by Neo-Oli** [comment]
I've been using it extensively for a week now and @BrainDamage also reported it to be working. I think it's ready to be merged. 

On a sidenote: proxychains and tsocks are both affected by https://github.com/termux/termux-packages/issues/465, if you try to connect to the local ssh socks proxy.

File: packages/proxychains-ng/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/rofl0r/proxychains-ng
+TERMUX_PKG_DESCRIPTION="a preloader which hooks calls to sockets in dynamically linked programs and redirects it through one or more socks/http proxies. It is a kind of proxifier."
+TERMUX_PKG_VERSION=4.12
+TERMUX_PKG_MAINTAINER="Oliver Schmidhauser @Neo-Oli"
+TERMUX_PKG_SRCURL=https://github.com/rofl0r/proxychains-ng/releases/download/v${TERMUX_PKG_VERSION}/proxychains-ng-${TERMUX_PKG_VERSION}.tar.xz
+TERMUX_PKG_SHA256=482a549935060417b629f32ddadd14f9c04df8249d9588f7f78a3303e3d03a4e
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS=" --prefix=${TERMUX_PREFIX}"
```
Comment (stephengroat) [inline packages/proxychains-ng/build.sh]: this isn't needed, `build-package.sh` already inserts this in

Comment (Neo-Oli) [reply packages/proxychains-ng/build.sh]: Right, thanks. I removed it in https://github.com/termux/termux-packages/pull/1575/commits/c59fed3a98d1d1d309453da0d5ad7acf513468b8

**Comment by kuburan** [comment]
@Neo-Oli  it work in my phone,.  tested through orbot app, and running nmap. and also work with dynamic chains nor random chains setting.

**Comment by Neo-Oli** [comment]
I've used this for more than a month now without any problems and is ready to be merged.

**Comment by Auxilus** [comment]
Working on my device too

**Comment by xeffyr** [comment]
Package is now available in [unstable-packages](https://github.com/termux/unstable-packages) repository.
```
pkg install unstable-repo
pkg install proxychains-ng
```

**Pull request closed without merging**